### PR TITLE
Follow "More flexible cluster configuration".

### PR DESCRIPTION
**Pull request by ueshin**
### Description

Follows "More flexible cluster configuration" at dbt-labs/dbt-spark#467.

- Reuse `dbt-spark`'s implementation
- Remove the dependency on `databricks-cli`
- Internal refactorings

### CODE REVIEW

File: dbt/adapters/databricks/python_submissions.py [hunk 1]
```diff
@@ -1,75 +1,65 @@
...
-            raise dbt.exceptions.RuntimeException(str(e)) from e
+class DbtDatabricksAllPurposeClusterPythonJobHelper(
+    DbtDatabricksBasePythonJobHelper, AllPurposeClusterPythonJobHelper
+):
+    def check_credentials(self) -> None:
+        self.credentials.validate_creds()
+        if not self.cluster_id:
+            raise ValueError(
+                "Databricks `http_path` or `cluster_id` of all-purpose cluster is required "
+                "for `all_purpose_cluster` submission method."
```
Comment (allisonwang-db) [inline dbt/adapters/databricks/python_submissions.py:64]:
```suggestion
                "Databricks `http_path` or `cluster_id` of an all-purpose cluster is required "
                "for the `all_purpose_cluster` submission method."
```

File: dbt/adapters/databricks/python_submissions.py [hunk 2]
```diff
@@ -1,75 +1,65 @@
...
+        self.auth_header.update({"User-Agent": user_agent, **http_headers})
 
-    def submit(self, compiled_code: str) -> None:
-        cluster_id = self.cluster_id
+    @property
+    def cluster_id(self) -> Optional[str]:  # type: ignore[override]
+        return self.parsed_model.get(
+            "cluster_id",
+            self.credentials.extract_cluster_id(
+                self.parsed_model.get("http_path", self.credentials.http_path)
```
Comment (allisonwang-db) [inline dbt/adapters/databricks/python_submissions.py:40]: Can we specify an `http_path` that is different from the one in dbt profiles for a Python model?

**Review by allisonwang-db — APPROVED** [review]
This looks good! Thanks for the refactoring.

File: dbt/adapters/databricks/python_submissions.py (hunk 2 above)
Comment (ueshin) [reply dbt/adapters/databricks/python_submissions.py:40]: Thanks for the heads-up.
Actually I double-checked it and found there is a mistake. I'll update it.

After the update, yes, we can specify `http_path` in the Python model config that is different from the one in the profile.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**